app/lib/actions/katello/content_view_environment/destroy.rb

```diff
@@ -34,6 +40,16 @@ def finalize
           end
           ::Katello::DockerMetaTag.cleanup_tags if input[:docker_cleanup]
         end
+
+        def remove_activation_key_associations(cv_env)
+          cv_env.activation_keys.each do |key|
+            key.content_view_environment_activation_keys.find_by(content_view_environment: cv_env)&.destroy
```

Comment:
This should use `ActivationKey#content_view_environments=` instead:

```suggestion
            key.content_view_environments = key.content_view_environments - cv_env
```

The ActivationKey model redefines the `content_view_environments=` method here: https://github.com/Katello/katello/blob/master/app/models/katello/activation_key.rb#L92

This ensures that the priority gets recalculated properly.

Reply:
Ack... I will make the change and push it.